src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationRow/integrationDirectoryApplicationRow.tsx

```diff
@@ -0,0 +1,186 @@
+import React from 'react';
+import {Link} from 'react-router';
+import capitalize from 'lodash/capitalize';
+import omit from 'lodash/omit';
+import styled from '@emotion/styled';
+import PropTypes from 'prop-types';
+
+import SentryTypes from 'app/sentryTypes';
+import {PanelItem} from 'app/components/panels';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import CircleIndicator from 'app/components/circleIndicator';
+import PluginIcon from 'app/plugins/components/pluginIcon';
+import {Organization, SentryApp, SentryAppInstallation} from 'app/types';
+import theme from 'app/utils/theme';
+
+const INSTALLED = 'Installed';
+const NOT_INSTALLED = 'Not Installed';
+const PENDING = 'Pending';
+
+type Props = {
+  app: SentryApp;
+  organization: Organization;
+  install?: SentryAppInstallation;
+  isOnIntegrationPage: boolean;
+  ['data-test-id']?: string;
+};
+
+export default class SentryApplicationRow extends React.PureComponent<Props> {
+  static propTypes = {
+    app: SentryTypes.SentryApplication,
+    organization: SentryTypes.Organization.isRequired,
+    install: PropTypes.object,
+    isOnIntegrationPage: PropTypes.bool,
+  };
+
+  get isInternal() {
+    return this.props.app.status === 'internal';
+  }
+
+  hideStatus() {
+    //no publishing for internal apps so hide the status on the developer settings page
+    return this.isInternal && !this.props.isOnIntegrationPage;
+  }
+
+  renderStatus() {
+    const {app, isOnIntegrationPage} = this.props;
+    const isInternal = this.isInternal;
+    const status = this.installationStatus;
+    if (this.hideStatus()) {
+      return null;
+    }
+    if (isOnIntegrationPage) {
+      //if internal and we show installation status, we don't show the learn more
+      if (isInternal) {
```

Comment:
@NisanthanNanthakumar Don't think we need this if condition anymore. I think we can always just return 
```
        <React.Fragment>
          <StatusIndicator status={status} isInternal={isInternal} />
        </React.Fragment>
```